sdk/azcore/internal/mock/mock.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package mock
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+)
+
+// Server is a wrapper around an httptest.Server.
+type Server struct {
+	srv    *httptest.Server
+	static *mockResponse
+	resp   []mockResponse
+	req    int
```

Comment:
Some comments would be good; I don't know what all these fields are for.
Also, do some of them (like req) need to be thread safe?

Reply:
Good point I'll add some comments here on the next round of test implementation.